### PR TITLE
Search url facets

### PR DESCRIPTION
**Pull request by danthewolfe**
* Facets now toggle and indicate if they are selected
 * if view is in url that view is used
 * &view=list or &view=grid are appended to url if they're not there on search results (page does not reload)
 * If an invalid or no view is specified it falls back to what is set on localstorage. if no view is specified there, grid is defaulted to.

### CODE REVIEW

File: app/assets/javascripts/components/layout/CollectionPageHeader.jsx [hunk 1]
```diff
@@ -50,7 +50,7 @@ var CollectionPageHeader = React.createClass({
       window.location.href = this.aboutUrl(this.props.collection);
     } else if (tab.props.value == "search") {
       window.location.href = this.browseUrl(this.props.collection);
-    } else {
+    } else{
```
Comment (jonhartzler) [inline app/assets/javascripts/components/layout/CollectionPageHeader.jsx]: we should probably remove this line

File: app/assets/javascripts/components/pages/Search/SearchControls.jsx [hunk 2]
```diff
@@ -48,15 +43,62 @@ var SearchControls = React.createClass({
...
+    EventEmitter.emit('SetGridList', view);
+  },
+
+  addViewToUrl: function(view) {
+    var searchString = window.location.search;
+    var regex = /\S+&view=/;
+    var garbage;
+    if(searchString.indexOf('&view=') > -1) {
+      garbage = searchString.replace(regex, '');
+      console.log(garbage);
```
Comment (jonhartzler) [inline app/assets/javascripts/components/pages/Search/SearchControls.jsx]: console.log

**Comment by jonhartzler** [comment]
:+1:

**Comment by jgondron** [comment]
:+1:

**Pull request merged**